clients/go/go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/camunda/zeebe/clients/go/v8
 
-go 1.19
+go 1.21
```

Comment:
Reverted it now. Makes sense to have it in a different PR. 👍 

I suspect that 1.17 to 1.19 was a bit special because of the generics. Usually, updating the Go version does not break anything (not sure of course what tools are used in the CI).